### PR TITLE
Implement new `EntityListLoader` that uses RTK Query hooks

### LINKED ISSUE #44492: Implement new `EntityListLoader` that uses RTK Query hooks
Make it work for 1 or 2 entities: databases and/or tables.

There is a separate issue to port the rest of entities.

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #44492

### Description

Similar to #44523 but for lists.
For now it is enabled for databases only.

Pagination-related things (`paginationState` HOC + `page`, `pageSize`, `onChangeHasMorePages` props) were not migrated here - there is only 1 usage. It will be refactored [later](https://github.com/metabase/metabase/issues/50324) when migrating the Users lists.

### How to verify

CI is green.

### CODE REVIEW

File: frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+  if (isObject(fetched) && fetched.data) {
+    const { data, ...rest } = fetched;
+    results = data;
+    metadata = rest;
+  } else {
+    results = fetched;
+  }
+
+  return { results, metadata };
+};
```
Comment (romeovs) [inline frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx:83]: Nit, I prefer using simple guards here.

```suggestion
const transformResponse = (fetched: unknown) => {
  if (!isObject(fetched) || !fetched.data) {
    return { data: fetched, metadata: {} }
  }   
  
  const { data, ...metadata } = fetched;
  return { data, metadata };
};
```

File: frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+    }
+  }, [data, onLoadedRef]);
+
+  // merge props passed in from stacked Entity*Loaders:
+  const allError = error || (allErrorProp == null ? null : allErrorProp);
+  const allFetched =
+    fetched && (allFetchedProp == null ? true : allFetchedProp);
+  const allLoaded = loaded && (allLoadedProp == null ? true : allLoadedProp);
+  const allLoading =
+    loading || (allLoadingProp == null ? false : allLoadingProp);
```
Comment (romeovs) [inline frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx:279]: Not sure what the exact semantics are here, but I think this is clearer:

```suggestion
  const allError = error || (allErrorProp ?? null);
  const allFetched = fetched && (allFetchedProp ?? true);
  const allLoaded = loaded && (allLoadedProp ?? true);
  const allLoading = loading || (allLoadingProp ?? false);
```

File: frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+}
+
+/**
+ * @deprecated HOCs are deprecated
+ */
+export const entityListLoaderRtkQuery =
+  (eolProps: any) =>
+  (ComposedComponent: (props: any) => ReactNode) =>
+  // eslint-disable-next-line react/display-name
+  (props: any): ReactNode => (
```
Comment (romeovs) [inline frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx:326]: Should we give this component a name and get rid of the `eslint-disable-next-line` directive?

File: frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+
+  return children;
+}
+
+/**
+ * @deprecated HOCs are deprecated
+ */
+export const entityListLoaderRtkQuery =
+  (eolProps: any) =>
+  (ComposedComponent: (props: any) => ReactNode) =>
```
Comment (romeovs) [inline frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx:324]: Is there a reason we're use `any` here? Can we not type this wrapper?

**Review by romeovs — APPROVED** [review]
A couple of nitpicks but otherwise looks fine!

File: frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx (hunk 1 above)
Comment (kamilmielnik) [reply frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx]: 329d2aaf25bdfc66260409de9047523f68aed949

File: frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx (hunk 2 above)
Comment (kamilmielnik) [reply frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx]: 0118847ed8fe95705a88f617f048de2359970a59

File: frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx (hunk 4 above)
Comment (kamilmielnik) [reply frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx]: f9d36da98f9cb0d07d9774d15e823bd66d82e561

File: frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx (hunk 3 above)
Comment (kamilmielnik) [reply frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx]: 828292b2b3aeb159fcf43ee4b5b5fa8ac3dfb325

**Pull request merged**